contracts/test-utils/src/assertions.ts

```diff
@@ -160,7 +161,8 @@ export async function expectTransactionFailedWithoutReasonAsync(p: sendTransacti
  * otherwise resolve with no value.
  */
 export async function expectContractCallFailedAsync<T>(p: Promise<T>, reason: RevertReason): Promise<void> {
-    return expect(p).to.be.rejectedWith(reason);
+    const rejectionMessageRegex = new RegExp(`^VM Exception while processing transaction: revert ${reason}$`);
```

Comment:
I don't think this will work in Geth. We have disabled Geth tests for now (see https://github.com/0xProject/0x-monorepo/pull/1754 and https://github.com/0xProject/0x-monorepo/pull/1775). Going forward, instead of running all of our unit tests through Geth and relying on features that don't exist upstream, we want to have integration tests that operate at a higher level and don't rely on any extra features.

Anyways, I'm okay with this for now but if we write integration tests for Geth in the future we'll have to update it.